### ISSUE TITLE
Setup auto-deployment for ansible switches

### ISSUE THREAD

**Issue by hakasapl** [CCI-MOC/ops-issues#1160]
We should setup a VM or container task that will apply manifests from [ansible switches](https://github.com/CCI-MOC/ansible-switches/) to our switches whenever the repository is updated.

Requirements:
* Every time the repo is updated, an ansible task should fire **only on the devices that manifests were changed**
* Whenever a switch update is pushed, the running config should be backed up, and also any commands that were run on the node
* Webhook integration to report anytime a job completed or failed to apply on a switch (Slack)

Initially, this will be a script written on a local machine. The program should determine when there is a new commit on the main brance of ansible-switches and invoke an ansible run.
* If a host var manifest was updated, you only need to run ansible on that switch using `--limit`
* If the vlan manifest was updated, the update needs to be run on every switch

You can test your code by running it on oct-head. It would be good to containerize this and run it as a container job that will be deployed on MOC infra cluster once it is running. You can use minikube or similar to write and test kubernetes manifests in the meantime.

Use sshuttle to access the switch management subnet and test ansible on.

Use OCT5-SW-TORS-A and OCT5-SW-TORS-B for dev work, those switches are not supporting anything production

Update:

- Program code for this has been written and tested locally.
- Next steps would be to containerize and test.

**Comment by joachimweyl** [comment]
@hakasapl a few questions on this issue.
1. Would this benefit from being broken into multiple issues?
   1. Update script creation
   1. Update script Testing
   1. Automation script creaiton
   1. Automation script testing
   1. etc.
1. Have we started on this?
   1. If not let's push this out to the next sprint

**Comment by hakasapl** [comment]
This is already well under way thanks to @tssala23 . The script is for the most part done as far as I know, I will make a separate issue for containerizing it.

**Comment by joachimweyl** [comment]
@tssala23 Please update this issue description with what has been completed and what the next steps are.

**Comment by joachimweyl** [comment]
@tssala23 what are the next steps for this issue? If you do not plan to finish it this sprint please break it down into what you will finish this sprint and what will be for the next sprint.

**Comment by larsks** [comment]
@joachimweyl there is a pull request (#1) that addresses this issue currently under review.

@tssala23  You should update your commit message and the PR description to include a link to the issue with which it is associated. This will update the issue to include a link to the commit (for mention in the commit message) and to this PR (for the mention in the PR description).

It's generally good practice for your commit message to provide some context for the changes.

If you write something like:

```
Closes: https://github.com/CCI-MOC/ops-issues/issues/1160
```

You can shorten that to:

```
Closes: cci-moc/ops-issues#1160
```

Then the referenced issue will be linked to the pr and will be closed automatically when this PR merges. If the PR and the issue are in the same repository, you can write:

```
Closes: #1160
```

(But that's not the case for this issue and pr.)

**Comment by hakasapl** [comment]
Closing this thanks to [this PR](https://github.com/CCI-MOC/ansible-switches-auto-deploy/pull/2) from @tssala23. Thank you! Next step is k8s manifests, which we will make a separate issue for.
